sulong/tests/com.oracle.truffle.llvm.tests/src/com/oracle/truffle/llvm/tests/interop/CxxMethodsTest.java

```diff
@@ -0,0 +1,199 @@
+            testLibrary.invokeMember("setY", point2, 8);
+            checkPoint(point, 3, -4);
+            checkPoint(point2, -6, 8);
+            // swap point <-> point2
+            swap.execute(point, point2);
+            checkPoint(point2, 3, -4);
+            checkPoint(point, -6, 8);
+            // calculate distance
+            Value distanceResult1 = squaredEuclideanDistance.execute(point, point2);
+            Value distanceResult2 = squaredEuclideanDistance.execute(point2, point);
+            Value distanceResult3 = point2.invokeMember("squaredEuclideanDistance", point);
+            Value distanceResult4 = point.invokeMember("squaredEuclideanDistance", point2);
+
+            Assert.assertEquals("distance(p, p2)", 0, Double.compare(225, distanceResult1.asDouble()));
+            Assert.assertEquals("distance(p2, p)", 0, Double.compare(225, distanceResult2.asDouble()));
+            Assert.assertEquals("p.distance(p2)", 0, Double.compare(225, distanceResult3.asDouble()));
+            Assert.assertEquals("p2.distance(p)", 0, Double.compare(225, distanceResult4.asDouble()));
+
+        } finally {
+            testLibrary.invokeMember("freeNativePoint", point);
+            freePoint.execute(point2);
+        }
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testNonExistingMethod() {
+        Value point = allocPoint.execute();
+        try {
+            point.invokeMember("methodWhichDoesNotExist");
+        } finally {
+            testLibrary.invokeMember("freeNativePoint", point);
+        }
+    }
+
+    @Test
+    public void testOverloadedMethods() {
+        Value xPoint = allocXtendPoint.execute();
+        try {
+            xPoint.invokeMember("setZ", 1);
+            Assert.assertEquals("getZ()", 1, xPoint.invokeMember("getZ").asInt());
+            Assert.assertEquals("getZ(2)", 3, xPoint.invokeMember("getZ", 2).asInt());
+        } finally {
+            freeXtendPoint.execute(xPoint);
+        }
+    }
+
+    @Test
+    public void testInheritedMethodsFromSuperclass() {
+        Value xPoint = allocXtendPoint.execute();
+        try {
+            xPoint.invokeMember("setX", 6);
+            xPoint.invokeMember("setY", 7);
+            Assert.assertEquals("direct call: getX()", 12, xPoint.invokeMember("getX").asInt());
+            Assert.assertEquals("superclass::getY()", 7, xPoint.invokeMember("getY").asInt());
+        } finally {
+            freeXtendPoint.execute(xPoint);
+        }
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testWrongArity() {
+        Value point = allocPoint.execute();
+        try {
+            constructor.execute(point);
+            point.invokeMember("setX", 6, 9, 4);
+        } finally {
+            freePoint.execute(point);
+        }
+    }
+
+    @Test
+    public void testConstructor() {
+        Value point = allocPoint.execute();
+        try {
+            constructor.execute(point);
```

Comment:
Not sure I understand this test. You're passing a point to the constructor? Shouldn't the constructor have zero arguments, and return a `Point`?

Reply:
True, it does indeed not make sense to pass the parameter here. I will change it to `Value point = constructor.newInstance()` and also remove the `execute` call of `allocPoint`.